puppet/hiera/hiera.yaml

```diff
@@ -1,13 +1,21 @@
 ---
 :backends:
-        - yaml
+  - yaml
 :yaml:
-        :datadir: '/usr/local/puppet/hiera/'
+  :datadir: '/usr/local/puppet/hiera/'
 :hierarchy:
-        - "nodes/external/secret/%{::fqdn}"
-        - "nodes/external/%{::fqdn}"
-        - "nodes/vagrant-%{::vagrant_extra_prefix}%{::is_vagrant}"
-        - "nodes/common"
-        - "vagrant-%{::is_vagrant}"
-        - "common"
-:merge_behavior: deeper
+  # things in nodes/ are part of event-specific repositories, not included in this repo
+  - "nodes/external/secret/%{::fqdn}"
+  - "nodes/external/secret/event-%{::event_name}"
+  - "nodes/external/secret/%{::environment}"
+  - "nodes/external/%{::fqdn}"
+  - "nodes/vagrant-%{::event_name}-%{::is_vagrant}"
+  - "nodes/vagrant-%{::is_vagrant}"
+  - "nodes/event-%{::event_name}"
+  - "nodes/%{::environment}"
+  - "nodes/common"
+  # things below nodes are included in this repo and visible to any event running uber
+  - "vagrant-%{::is_vagrant}"
+  - "%{::environment}"
+  - "common"
+# :merge_behavior: deeper # used to need this, probably not anymore.
```

Comment:
This all looks pretty good, but keep in mind that we'll want to override some settings in the staging server that don't qualify as secret. Like if we want to test dealer registration before it opens, we need to override those dates. Because of that, I'm still not sure it makes a ton of sense to put the environment YAML above the event YAML.